### PR TITLE
interface network option improvement

### PR DESCRIPTION
**Pull request by liangxin1300**
-  Give valid value list for the -i option when the value is invalid
```
# crm cluster init -i xxx
ERROR: cluster.init: Invalid value 'xxx' for -i/--interface option

# After applying this PR

# crm cluster init -i xxx
ERROR: cluster.init: Invalid value 'xxx' for -i/--interface option, should be enp1s0, enp7s0, enp8s0 or 192.168.122.189, 10.10.10.61, 20.20.20.61
```
-  On join side, adjust the condition of comparing the link number
```
# on init node
# crm cluster init -i 10.10.10.61 -i 20.20.20.61 -y

# on join node
# crm cluster join
# ...
# ERROR: cluster.join: knet transport of all cluster nodes need 2 links via '-i' options, but provided 1

# After applying this PR

# crm cluster join
# ...
# Address for ring0 [192.168.122.145]
# Address for ring1 []10.10.10.62
```

- Minor refactoring of the get_address_list function
```
# no way to add another ring on interactive mode
# crm cluster init
...
Address for ring0 [192.168.122.189]
INFO: Configure SBD:

# After applying this PR
# crm cluster init
...
Address for ring0 [192.168.122.145]

Add another ring (y/n)?
```

```
# no need to ask for confirm for each -i option
# crm cluster init -i enp7s0 -i enp8s0
...
Address for ring0 [10.10.10.61]

Add another ring (y/n)?

# After applying this PR
 # crm cluster init -i enp7s0 -i enp8s0
...
Address for ring0 [10.10.10.62]
Address for ring1 [20.20.20.62]
INFO: Configure SBD:
```

### CODE REVIEW

File: crmsh/bootstrap.py [hunk 1]
```diff
@@ -1325,16 +1325,23 @@ def get_address_list() -> typing.List[str]:
...
+        loop_count = len(_context.default_ip_list)
+    else:
+        # non-interative mode
+        # will ask for each interface, max is Context.MAX_LINK_NUM
+        loop_count = min(Context.MAX_LINK_NUM, len(_context.interfaces_inst.nic_list))
+
     ringXaddr_list = []
-    loop_count = min(Context.MAX_LINK_NUM, len(_context.default_ip_list))
     for i in range(loop_count):
+        # at non-interactive mode, need to confirm after default ip list been consumed
```
Comment (nicholasyang2022) [inline crmsh/bootstrap.py:1337]: Should be "at interactive mode"?

**Review by nicholasyang2022 — COMMENTED** [review]
> should be ['enp1s0', 'enp7s0', 'enp8s0'] or ['192.168.122.189', '10.10.10.61', '20.20.20.61']

This is confusing. How about

```
should be 'enp1s0', 'enp7s0', 'enp8s0', '192.168.122.189', '10.10.10.61', or '20.20.20.61'.
```

**Comment by nicholasyang2022** [comment]
It does not make sense to ask for another ring when `-i` is specified in cmdline, especially when multiple `-i` is specified. Or at least, the question should have a default answer 'N' so that the user can simply skip the question by pressing Enter.

**Comment by liangxin1300** [comment]
> It does not make sense to ask for another ring when `-i` is specified in cmdline, especially when multiple `-i` is specified. Or at least, the question should have a default answer 'N' so that the user can simply skip the question by pressing Enter.

Changed, not ask when -i was specified

File: crmsh/bootstrap.py [hunk 2]
```diff
@@ -1325,15 +1325,23 @@ def get_address_list() -> typing.List[str]:
     else:
         valid_func = Validation.valid_ucast_ip
 
+    if _context.yes_to_all or _context.nic_addr_list:
+        loop_count = len(_context.default_ip_list)
+    else:
+        # interative mode or without -i option specified
```
Comment (nicholasyang2022) [inline crmsh/bootstrap.py:1331]:
```suggestion
        # interative mode and without -i option specified
```

**Review by nicholasyang2022 — APPROVED** [review]

**Pull request merged**